spec/migrations/20171025122732_move_ems_refresh_args_to_data_spec.rb

```diff
@@ -0,0 +1,88 @@
+require_migration
+
+describe MoveEmsRefreshArgsToData do
+  let(:miq_queue_stub) { migration_stub(:MiqQueue) }
+  let(:targets) { [['Vm', 1], ['Host', 2]] }
+  let(:refresh_queue_options) do
+    {
+      :class_name  => 'EmsRefresh',
+      :method_name => 'refresh',
+      :role        => 'ems_inventory',
+      :queue_name  => 'ems_1',
+      :zone        => 'default',
+    }
+  end
+  let(:refresh_new_target_queue_options) do
+    {
+      :class_name  => 'EmsRefresh',
+      :method_name => 'refresh_new_target',
+      :role        => 'ems_inventory',
+      :queue_name  => 'ems_1',
+      :zone        => 'default',
+    }
+  end
+
+  migration_context :up do
+    it "Moves EmsRefresh.refresh queue args to data" do
+      q_item = miq_queue_stub.create!(refresh_queue_options.merge(:args => [targets]))
+
+      migrate
+
+      expect(Marshal.load(q_item.reload.msg_data)).to match_array(targets)
+    end
+
+    it "Leaves EmsRefresh.refresh args empty" do
+      q_item = miq_queue_stub.create!(refresh_queue_options.merge(:args => [targets]))
+
+      migrate
+
+      expect(q_item.reload.args).to match_array([])
+    end
+
+    it "Ignores unrelated queue items" do
+      args = [{:ems_ref => "vm-123"}]
+      q_item = miq_queue_stub.create!(refresh_new_target_queue_options.merge(:args => args))
+
+      migrate
+
+      expect(q_item.reload.args).to match_array(args)
+    end
+  end
+
+  migration_context :down do
+    it "Moves EmsRefresh.refresh data to args" do
+      q_item = miq_queue_stub.create!(refresh_queue_options.merge(:msg_data => Marshal.dump(targets)))
+
+      migrate
+
+      expect(q_item.reload.args).to match_array(targets)
+    end
+
+    it "If there are no args" do
+      q_item = miq_queue_stub.create!(refresh_queue_options.merge(:msg_data => nil))
+
+      migrate
+
+      expect(q_item.reload.args).to match_array([])
+    end
+
+    it "Handle invalid marshal format errors" do
+      # "\x03\x00\x00" is an empty array in Marshal version 3.0
```

Comment:
Is there a version of this for the up migration?

Reply:
I wasn't sure how to test `Marshal.dump` failing due to a version issue, ideas?

Reply:
Let me try putting an anonymous class in the args, that should do it.

Reply:
Scratch that, that won't even serializing the value to args

Reply:
Oh I see...it's only the load part that fails.  :+1: